src/reconcile/rec_write.c

```diff
@@ -294,6 +294,8 @@ __reconcile(WT_SESSION_IMPL *session, WT_REF *ref, WT_SALVAGE_COOKIE *salvage, u
       F_ISSET(r, WT_REC_EVICT) && !WT_PAGE_IS_INTERNAL(r->page) && r->multi_next == 1 &&
       F_ISSET(r, WT_REC_CALL_URGENT) && !r->update_used && r->cache_write_restore) {
         WT_STAT_CONN_DATA_INCR(session, cache_eviction_blocked_no_progress);
+        if (F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT))
```

Comment:
Can we add a comment here? Saying "If eviction does not make any progress then we set this flag so that application threads refresh the snapshot to evict the latest content."

Reply:
Good idea - done in 1e051cb.